sdk/go/common/resource/plugin/langruntime_plugin.go

```diff
@@ -55,35 +56,125 @@ type langhost struct {
 // plugin could not be found, or an error occurs while creating the child process, an error is returned.
 func NewLanguageRuntime(host Host, ctx *Context, runtime, workingDirectory string, info ProgramInfo,
 ) (LanguageRuntime, error) {
-	path, err := workspace.GetPluginPath(ctx.Diag,
-		apitype.LanguagePlugin, strings.ReplaceAll(runtime, tokens.QNameDelimiter, "_"), nil, host.GetProjectPlugins())
+	attachPort, err := GetLanguageAttachPort(runtime)
 	if err != nil {
 		return nil, err
 	}
 
-	contract.Assertf(path != "", "unexpected empty path for language plugin %s", runtime)
+	var plug *plugin
+	var client pulumirpc.LanguageRuntimeClient
+	if attachPort != nil {
+		port := *attachPort
 
-	args, err := buildArgsForNewPlugin(host, info.RootDirectory(), info.Options())
-	if err != nil {
-		return nil, err
-	}
+		handshake := func(
+			ctx context.Context, bin string, prefix string, conn *grpc.ClientConn,
+		) (*pulumirpc.LanguageHandshakeResponse, error) {
+			req := &pulumirpc.LanguageHandshakeRequest{
+				EngineAddress:    host.ServerAddr(),
+				RootDirectory:    filepath.Dir(bin),
+				ProgramDirectory: filepath.Dir(bin),
```

Comment:
Genuinely unsure here -- should these be the same?

Reply:
Actually when attaching we can simply leave them blank like provider does, and set them both when we spawn it, and this is what provider does as well.

Reply:
As commented these _should_ be blank